tests/Pregel/ConductorStateTest.cpp

```diff
@@ -0,0 +1,210 @@
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Heiko Kernbach
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Pregel/Conductor/State.h"
+#include "Pregel/PregelOptions.h"
+#include "Pregel/Conductor/ExecutionStates/CreateWorkers.h"
+
+#include "Basics/ResultT.h"
+
+#include <gtest/gtest.h>
+
+using namespace arangodb::pregel;
+using namespace arangodb::pregel::conductor;
+
+static const std::string databaseName = "dontCare";
+
+struct LookupInfoMock : conductor::LookupInfo {
+  LookupInfoMock(std::vector<std::string> serverIDs) : _servers(serverIDs) {}
+  ~LookupInfoMock() = default;
+
+  [[nodiscard]] auto getServerMapVertices() const -> ServerMapping override {
+    ServerMapping mapping{};
+    for (auto const& serverID : _servers) {
+      mapping.emplace(serverID,
+                      std::map<std::string, std::vector<std::string>>{});
+    }
+    return mapping;
+  }
+
+  // edges related methods:
+  [[nodiscard]] auto getServerMapEdges() const -> ServerMapping override {
+    ServerMapping mapping{};
+    for (auto const& serverID : _servers) {
+      mapping.emplace(serverID,
+                      std::map<std::string, std::vector<std::string>>{});
+    }
+    return mapping;
+  }
+
+  // both combined (vertices, edges) related methods:
+  [[nodiscard]] auto getAllShards() const -> ShardsMapping override {
+    return {};
+  }
+
+  [[nodiscard]] auto getCollectionPlanIdMapAll() const
+      -> CollectionPlanIDMapping override {
+    return {};
+  }
+
+  std::vector<std::string> _servers;
+};
+
+TEST(ConductorStateTest,
+     must_always_be_initialized_with_initial_execution_state) {
+  std::vector<std::string> emptyServers{};
+  auto cState = ConductorState(ExecutionSpecifications(),
+                               std::make_unique<LookupInfoMock>(emptyServers));
+  ASSERT_EQ(cState._executionState->name(), "initial");
+}
+
+TEST(CreateWorkersStateTest, creates_as_many_messages_as_required_servers) {
+  std::vector<std::vector<std::string>> amountOfServers = {
+      {}, {"ServerA"}, {"ServerA", "ServerB"}};
+
+  for (auto const& servers : amountOfServers) {
+    auto cState = ConductorState(ExecutionSpecifications(),
+                                 std::make_unique<LookupInfoMock>(servers));
+    auto createWorkersState = CreateWorkers(cState);
+    auto msgs = createWorkersState.messages();
+    ASSERT_EQ(msgs.size(), servers.size());
+
+    for (auto const& serverName : servers) {
+      ASSERT_TRUE(msgs.contains(serverName));
+    }
+  }
+}
+
+TEST(CreateWorkersStateTest, check_that_conductor_creates_all_worker_pids) {
```

Comment:
why "check"? 
It is more: creates worker pids from received messages

Reply:
Thank you, applied suggestion.